packages/containers-shared/src/types.ts

```diff
@@ -11,24 +11,53 @@ export type BuildArgs = {
 	tag: string;
 	pathToDockerfile: string;
 	/** image_build_context or args.PATH. if not provided, defaults to the dockerfile directory */
-	buildContext?: string;
+	buildContext: string;
 	/** any env vars that should be passed in at build time */
 	args?: Record<string, string>;
 	/** platform to build for. defaults to linux/amd64 */
 	platform?: string;
 	/** sets --network=host at build time. only used by workers CI. */
 	setNetworkToHost?: boolean;
 };
+export type DockerfileConfig = SharedContainerConfig & {
+	/**  absolute path, resolved relative to the wrangler config file */
+	dockerfile: string;
+	/** absolute path, resolved relative to the wrangler config file. defaults to the directory of the dockerfile */
+	image_build_context: string;
+	image_vars?: Record<string, string>;
+};
+export type RegistryLinkConfig = SharedContainerConfig & {
+	registry_link: string;
+};
+
+export type ContainerNormalisedConfig = RegistryLinkConfig | DockerfileConfig;
+export type InstanceTypeOrLimits =
+	| {
+			/** if undefined in config, defaults to instance_type */
+			disk_size?: number;
```

Comment:
```suggestion
			disk_mb?: number;
```
just so this is slightly more consistent with the naming elsewhere